### DISCUSSION TITLE
TtkBootstrap Tableview sort problem ?

### DISCUSSION THREAD

**Discussion by AbdoMusk** [israel-dryer/ttkbootstrap#390] [Q&A] [answered]
Hello, I'm new to Ttkbootstrap, I liked that some of the new features that Ttkbootstrap provides that don't exist on Tkinter; one of those is Tableview, I wanted to display some data on a Tableview it works fine but when I try to sort the data I get an error, here's my working **environment:**

Ttkbootstrap Version: 1.10.0
Python 3.10.8
Windows 10

**Error:**
![image](https://user-images.githubusercontent.com/31053920/208643348-7f8e2c49-8fe3-4c4f-9943-81cabbba7e9f.png)

After digging into the error I found out that self.view.configure("displaycolumns") in ttkbootstrap.tableview.py returns None

**My code** (I tested with plain tuple too):

```
import ttkbootstrap as ttk
from ttkbootstrap.tableview import Tableview
from ttkbootstrap.constants import *
from collections import namedtuple

OfferItem = namedtuple('OfferItem', ['type', 'name', 'price'])


class MainWindow:
    def __init__(self):
        self.app = ttk.Window(themename='superhero')

        self.coldata = [
            {"text": "Type", "stretch": False},
            "Name",
            { "text": "Price", "stretch": False},
        ]

        self.dt = Tableview(
            master=self.app,
            coldata=self.coldata,
            rowdata=[
        OfferItem(type='Test', name='TestName', price=7911.38),
        OfferItem(type='Test2', name='TestName2', price=3500.00),
    ],
            paginated=True,
            searchable=True,
            bootstyle=PRIMARY,
            stripecolor=(self.app.style.colors.light, None),
        )
        self.dt.pack(fill=BOTH, expand=YES, padx=10, pady=10)
```

**The solution I found:**

is to change cols = self.view.configure("displaycolumns") to cols = self.view["columns"] in ttkbootstrap.tableview.py
![image](https://user-images.githubusercontent.com/31053920/208646385-6b675773-9828-4268-91c8-acb6cc3a1cae.png)
![image](https://user-images.githubusercontent.com/31053920/208645966-00dca7b3-b269-4e0e-adae-e2c990e338b7.png)

I'm not sure if that could be a pull request so I just put it here as a show-and-tell discussion.

**Comment by AbdoMusk** [comment]
Also showing or hiding a column create an error, because the functions show() hide() uses self.view.configure("displaycolumns") and self.view.configure("columns")

![image](https://user-images.githubusercontent.com/31053920/208652406-0beae44e-ec1d-40df-ad4b-2c8b1d5f05f9.png)

after changing it to self.view["displaycolumns"] and self.view["columns"] showing or hiding column in Tableview does work great!

**Reply by israel-dryer** [reply]
Thank you. This is on my radar now.

**Comment by israel-dryer** [comment, marked as answer]
Thanks for the detailed report and for digging into the source!

This was a real bug in `TableColumn.displayindex`. In 1.10.0 it read the columns with `self.view.configure("displaycolumns")` — but `ttk.Treeview.configure(option)` returns tkinter's option-spec tuple `(name, dbName, dbClass, default, current)`, not the value. So the `"#all"` guard didn't match and `.index(self.cid)` ran against that spec tuple, which is the error you hit.

It was fixed in **v1.18.0** by switching to `self.view.cget("displaycolumns")` (with the `#all` default handled explicitly), so upgrading to the current release resolves the sort error.

Your instinct was right — the shipped fix uses `.cget("displaycolumns")` rather than `["columns"]` so that hidden/reordered columns still sort correctly, but it's the same root cause you identified. 👍
